app/src/main/java/org/mozilla/fenix/onboarding/DefaultBrowserNotificationWorker.kt

```diff
@@ -0,0 +1,103 @@
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Context.NOTIFICATION_SERVICE
+import android.content.Intent
+import android.os.Build
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationCompat.BADGE_ICON_NONE
+import androidx.core.app.NotificationManagerCompat
+import androidx.core.content.ContextCompat
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+import mozilla.components.support.base.ids.SharedIdsHelper
+import org.mozilla.fenix.HomeActivity
+import org.mozilla.fenix.R
+import org.mozilla.fenix.utils.Settings
+
+class DefaultBrowserNotificationWorker(
+    context: Context,
+    workerParameters: WorkerParameters
+) : Worker(context, workerParameters) {
+
+    override fun doWork(): Result {
+        ensureChannelExists()
+        NotificationManagerCompat.from(applicationContext)
+            .notify(NOTIFICATION_TAG, NOTIFICATION_ID, buildNotification())
+        return Result.success()
+    }
+
+    /**
+     * Build the notification with site controls to be displayed while the web app is active.
+     */
+    private fun buildNotification(): Notification {
+        val channelId = ensureChannelExists()
+        val intent = Intent(applicationContext, HomeActivity::class.java)
+        intent.putExtra(INTENT_DEFAULT_BROWSER_NOTIFICATION, true)
+
+        val pendingIntent = PendingIntent.getActivity(
+            applicationContext,
+            SharedIdsHelper.getNextIdForTag(applicationContext, NOTIFICATION_PENDING_INTENT_TAG),
+            intent,
+            0
+        )
+
+        with(applicationContext) {
+            val appName = getString(R.string.app_name)
+            return NotificationCompat.Builder(this, channelId)
+                .setSmallIcon(R.drawable.ic_status_logo)
+                .setContentTitle(
+                    applicationContext.getString(R.string.notification_default_browser_text, appName))
+                .setBadgeIconType(BADGE_ICON_NONE)
+                .setColor(ContextCompat.getColor(this, R.color.primary_text_light_theme))
+                .setPriority(NotificationCompat.PRIORITY_LOW)
+                .setShowWhen(false)
+                .setContentIntent(pendingIntent)
+                .build()
+        }
+    }
+
+    /**
+     * Make sure a notification channel for the powered by notifications exists.
+     *
+     * Returns the channel id to be used for notifications.
+     */
+    private fun ensureChannelExists(): String {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            val notificationManager: NotificationManager =
+                applicationContext.getSystemService(NOTIFICATION_SERVICE) as NotificationManager
+
+            val channel = NotificationChannel(
+                NOTIFICATION_CHANNEL_ID,
+                applicationContext.getString(R.string.notification_default_browser_channel_name),
```

Comment:
Is it intentional to have a notification channel _only_ for the default browser notification or is there something more generic if/when we want to have more that fall under that grouping? I can see Brave has a generic "Brave Ads" [group ID](https://github.com/brave/brave-core/blob/1cb5818aa0b70666c6aeea5ea9c06cc4e712171a/android/java/org/chromium/chrome/browser/notifications/channels/BraveChannelDefinitions.java#L19).

Reply:
Sure, I'll update the channel name to `Marketing` to allow for more generic usage in the future.  Thanks,

Reply:
> Sure, I'll update the channel name to `Marketing` to allow for more generic usage in the future. Thanks,

Was this string something that the content/UX team verified? This will eventually be translated into various languages and deleting/updating this if we want to use channel grouping for something else will be annoying in the future.

Reply:
Yes, the original requirement does not have any text specified.  That's why I added @topotropic to review the text and she gave the approval above.